src/fmu/dataio/providers/_base.py

```diff
@@ -0,0 +1,18 @@
+from abc import ABC, abstractmethod
+
+from pydantic import BaseModel
+
+
+class Provider(ABC):
+    @abstractmethod
+    def get_metadata(self) -> BaseModel:
```

Comment:
It may be worth making a light base class wrapper as the return type just for my specificity, i.e. `MetadataModel`

Reply:
Not really needed.